lib/graphql/schema/field.rb

```diff
@@ -643,9 +643,7 @@ def resolve(object, args, query_ctx)
               inner_object = obj.object
 
               if defined?(@hash_key)
-                inner_object.fetch(@hash_key) {
-                  inner_object[@hash_key_str]
-                }
+                inner_object[@hash_key].nil? ? inner_object[@hash_key_str] : inner_object[@hash_key]
```

Comment:
Actually I am worried this might not work; if the value is actually `nil` this will revert to `@hash_key_str`

Reply:
Curious what're your thoughts @rmosolgo? Personally it's a bit weird to see `hash_key` not be guarded by `is_a?(Hash)` -- but maybe it's meant to be used this way as noticed in previous PRs.

Reply:
We're also unsure how `hash_key` is supposed to be used. Reference [in the docs](https://graphql-ruby.org/fields/introduction.html#field-resolution): `If the underlying object is a Hash, lookup a key in that hash.`. Is this not the case anymore?

Reply:
I chose `fetch` because it _would_ return `nil` if the key is present in `inner_object` and the value is `nil` -- I think we should retain that behavior.

How about:  

```suggestion
                inner_object.key?(@hash_key) ? inner_object[@hash_key] : inner_object[@hash_key_str]
```

(it looks like `.key?` is what the code _used_ to use: https://github.com/rmosolgo/graphql-ruby/pull/4015/files#diff-53ed7db7a95b84a82815af3a920ccd962fce9469ef21ed25111fb149bc1406eaL658-L661)

Reply:
Sorry if I'm unclear @rmosolgo or maybe we're mis-understanding, but actually I think the problem is that we have this:

```ruby
if defined?(@hash_key)
```

check outside of checking whether or not the underlying object is a `Hash`, example code from today: [link](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/field.rb#L645). So even if we use `.key?` I think it would break... maybe the proper fix is to actually move this code block back into [here](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/field.rb#L660)?